internal/epoch/epoch_manager_test.go

```diff
@@ -393,26 +393,26 @@ func TestIndexEpochManager_NoCompactionInReadOnly(t *testing.T) {
 	// Use assert.Eventually here so we'll exit the test early instead of getting
 	// stuck until the timeout.
 	var (
-		loadedDone bool
-		loadedErr  error
+		loadedDone atomic.Bool
+		loadedErr  atomic.Value
 	)
 
 	go func() {
-		defer func() {
-			loadedDone = true
-		}()
+		if err := te2.mgr.Refresh(ctx); err != nil {
+			loadedErr.Store(err)
+		}
 
-		loadedErr = te2.mgr.Refresh(ctx)
 		te2.mgr.backgroundWork.Wait()
+		loadedDone.Store(true)
 	}()
 
-	if !assert.Eventually(t, func() bool { return loadedDone }, time.Second*5, time.Second) {
```

Comment:
`loadDone` is being red here without any synchronization that ensures the value is propagated across goroutines.